app/models/ability.rb

```diff
@@ -76,9 +76,11 @@ def signed_in(user)
     can :manage, Event do |event|
       event.users.include?(user)
     end
-    # can create an event until the last day of a conference
-    can :create, Event, program_id: Conference.where('end_date >= ?', Date.today).map { |conference| conference.program.id}.compact
 
+    # can create an event if program has call for papers defined
+    can [:new, :create], Event do |event|
+      event.program.cfp_open? && event.new_record?
```

Comment:
Why do we need `&& event.new_record?` here too? And why don't we allow a user to create an Event only for his/her own user?